pkg/server/sotw/v3/server.go

```diff
@@ -227,7 +226,24 @@ func (s *server) process(str stream.Stream, reqCh <-chan *discovery.DiscoveryReq
 				return err
 			}
 
-			watches.responders[res.GetRequest().TypeUrl].nonce = nonce
+			// Track the resources returned in the response
+			// Those are staged pending the client ACK
+			// The protocol clearly states that if we send another response prior to an ACK
+			// the previous one is to be considered as discarded
```

Comment:
Can you link to the protocol section that states that? I don't find the wording and I'm not sure this concern is valid.

Not having to resend resources that were already sent would make the code much simpler, as you would not need to track aknowledgement, afaik (no concept of "pending" resource).

Reply:
OK, could you validate envoy's behavior in the case you describe? gRPC will cache versions sent at version N, and only update those that changed at N+1, so nothing is lost. I think it'd be tricky to even implement something different on the client side, but perhaps Envoy does.